openfe/utils/visualization.py

```diff
@@ -140,22 +157,49 @@ def draw_mapping(mol1_to_mol2: Dict[int, int],
     if not d2d:
         d2d = Chem.Draw.rdMolDraw2D.MolDraw2DCairo(600, 300, 300, 300)
 
-    # Need to squash to 2D
-    mol1_copy = copy.deepcopy(mol1)
-    mol2_copy = copy.deepcopy(mol2)
-    AllChem.Compute2DCoords(mol1_copy)
-    AllChem.Compute2DCoords(mol2_copy)
+    return _draw_molecules(
+        d2d,
+        [mol1, mol2],
+        atoms_list=atoms_list,
+        bonds_list=bonds_list,
+        atom_colors=atom_colors,
+        highlight_color=red,
+    )
 
-    # Use the d2d object we instantiated or the one passed in by the user
-    d2d.drawOptions().useBWAtomPalette()
-    d2d.drawOptions().continousHighlight = False
-    d2d.drawOptions().setHighlightColour(red)
-    d2d.drawOptions().addAtomIndices = True
-    d2d.DrawMolecules(
-        [mol1_copy, mol2_copy],
-        highlightAtoms=atoms_list,
-        highlightBonds=bonds_list,
-        highlightAtomColors=atom_colors,
+
+def draw_one_molecule_mapping(mol1_to_mol2, mol1, mol2, d2d=None):
+    """Draw the mapping visualization for a single molecular of a mapping
+
+    This will always draw ``mol1``. To draw ``mol2``, switch order/invert
+    ``mol1_to_mol2`` mapping.
+    """
+    uniques = _get_unique_bonds_and_atoms(mol1_to_mol2, mol1, mol2)
+    atoms_list = [uniques["atoms"] | uniques["elements"]]
+    bonds_list = [uniques["bonds"]]
+    red = (220/255, 50/255, 32/255, 1)
+    blue = (0, 90/255, 181/255, 1)
+
+    atom_colors = [{at: blue for at in uniques["elements"]}]
+
+    if d2d is None:
+        # TODO: we should try to be smarter about the default size here
+        d2d = Chem.Draw.rdMolDraw2D.MolDraw2DCairo(300, 300, 300, 300)
+
+    return _draw_molecules(d2d, [mol1], atoms_list, bonds_list, atom_colors,
+                           red)
+
+
+def draw_unhighlighted_molecule(mol, d2d=None):
```

Comment:
Do we want to add this as a method accessed by Molecule? (otherwise I'm not super sure what the intended use case is here)

Reply:
That would be an option, though I'm not going to implement it now. We could use that to get rdkit-like image reprs in notebooks.

The actual idea was that I originally wanted to draw the network with unhighlighted molecules as the nodes, and then when you select an edge, the node visualization changed to the highlighted form. In that case, there's obviously a need for this method. I'm not using it in the final version of my network visualizer, but I figured it could be useful in other cases.

Reply:
We could, or we could just wait until tomorrow when I'm asked to show that image either when you click on the node or when you hover over it (I'm not sure which one they'll ask for).